packages/components/src/spectrum/dateAndTime.ts

```diff
@@ -12,3 +18,15 @@ export {
   TimeField,
   type SpectrumTimeFieldProps as TimeFieldProps,
 } from '@adobe/react-spectrum';
+
+export type { CalendarDate, CalendarDateTime, ZonedDateTime };
+
+export type DateValue = CalendarDate | CalendarDateTime | ZonedDateTime;
+
+export type MappedDateValue<T> = T extends ZonedDateTime
```

Comment:
I would probably add a comment on these types describing why we need them.

Reply:
Pushed an update with some comments.